### DISCUSSION TITLE
GitHub Codespaces

### DISCUSSION THREAD

**Discussion by jeff-horton-ho-sas** [UKHomeOffice-test/docs-as-code-nunjucks#31] [Ideas]
Can [GitHub Codespaces](https://github.com/features/codespaces) or something like [GitPod](https://gitpod.io/) be used to provide an easier way for non-technical users to make changes to the repo in their browser?  Essentially gives you a preconfigured VS code + dev environment running in the brwoser, linked from a button in the repo. Is it possible to configure it in a way to hide the complexity from non-technical users? Is it worth the cost?

**Comment by robertdeniszczyc2** [comment]
There would be a lot of questions around this, but "configure it in a way to hide the complexity from non-technical users" is also a big one here. IDEs could be intimidating for a non-technical user, so could increase the barrier to contributions.

**Comment by edhamiltonHO** [comment]
In a similar vein I've also had a quick look into how editing files directly through the browser works on Github.com. In short it looks pretty simple to create a branch, add or amend files etc. [some more information on Github flow from just the browser](https://github.blog/2013-07-11-github-flow-in-the-browser/)

The big question is around being able to preview the impact of those changes, apparently there is some basic MD previewing. Nothing will get built (succesfully or not) until at least a PR is created, and then would only be a basic build test unless we define something more robust.

Do we need non-technical users to be able to build locally or via something like Codespaces, or can we just leave that to be sorted out by someone technical if/when they get involved later at the PR?

**Reply by willhiorns** [reply]
I'm on a locked-down machine at the moment where there's no IDE.  Going to another to do some in-IDE editing is a disincentive for lazy people like me.  However, I should probably try out what can be done via browser before I form any strong opinion.
